woocommerce/woocommerce
# Product Gallery (Beta): fix the image aspect-ratio when in full page mode

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/52947



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure to have a product with a main image and gallery images (nice to have some vertical and horizontal image)
2. Go to Editor > Single Product template
3. Replace the Product Image Gallery block with the Product Gallery (Beta) block.
4. Save and go to frontend product page
5. Click on the image to open full page view
6. Expected:
    - gallery occupies the full height
    - image preserves its aspect ratio
    - image uses most of the space (either width or height).
        - Exception for vertical images, it will be tackled in another issue: https://github.com/woocommerce/woocommerce/issues/47569
        - Images may reach their max size and in that case they should not be stretched to lose the quality
8. Check the same in mobile view as well as in the incognito mode without the admin bar

Examples:

Scenario | Before | After
-- | -- | --
Vertical image | <img width="921" alt="image" src="https://github.com/user-attachments/assets/8f7db3b6-5bdb-4450-a7d7-df1313240bc4"> | <img width="921" alt="image" src="https://github.com/user-attachments/assets/5a309e53-a69e-4ee6-9483-2205bf9abd7b">
Horizontal image |<img width="923" alt="image" src="https://github.com/user-attachments/assets/d82e5d2f-f907-49ca-9024-29e503892a1f"> | <img width="920" alt="image" src="https://github.com/user-attachments/assets/c04aa9de-9386-40db-8d25-4e44154cd71a">
Mobile view | <img width="646" alt="image" src="https://github.com/user-attachments/assets/ff458768-3c6c-4bd6-a37c-8635bd26e2ab"> | <img width="647" alt="image" src="https://github.com/user-attachments/assets/0929cbb8-3d59-4fcc-90f2-57576313e7b2">
Without admin bar | <img width="1588" alt="image" src="https://github.com/user-attachments/assets/11145bc1-74a0-4b4f-abae-6f57eb350b78"> | <img width="1057" alt="image" src="https://github.com/user-attachments/assets/365996f1-2e6b-4971-bf23-9522b3f4add2">




### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
